### PR TITLE
Add a injectJavaScript method to the WebView component

### PR DESCRIPTION
**Pull request by lukemiles**
Currently, < WebView > allows you to pass JS to execute within the view. This works great, but there currently is not a way to execute JS after the page is loaded. We needed this for our app.

We noticed that the WebView had messaging support added (see #9762) . Initially, this seemed like more than enough functionality for our use case - just write a function that's injected on initial load that accepts a message with JS, and `eval()` it. However, this broke once we realized that Content Security Policy can block the use of eval on pages. The native methods iOS provide to inject JS allow you to inject JS without CSP interfering. So, we just wrapped the native methods on iOS (and later Android) and it worked for our use case. The method injectJavaScript was born.

Now, after I wrote this code, I realized that #8798 exists and hadn't been merged because of a lack of tests. I commend what was done in #8798 as it sorely solves a problem (injecting JS after the initial load) and has more features than what I've done here (allows you to return the response). But, it uses `eval()` internally which *makes it subject to Content Security Policy*. 

I'm not exactly sure what to do, but I think something should be merged. Perhaps I can add a way to try to return the results without using eval() but that seems tricky. I think the best way would be some sort of internal check when the WebView is loaded to see if CSP applies, and to not return a result on injected JavaScript (disable the use of eval() ) if that is the case. Or, you could make the argument that you don't need an evaluateJavascript function because you can accomplish it using the messaging features as it would work in all of the same places.

Looking for input - thanks.

Code was tested on iOS but not yet on Android - setting up my emulator later on.

### CODE REVIEW

**Comment by ericnakagawa** [comment]
@lukemiles This looks like it could be a useful addition. I'm having someone from my team review.
@jacobp100 Thanks for reviewing, too.

**Comment by ericnakagawa** [comment]
@lukemiles Can you add some tests to this PR?

**Comment by lukemiles** [comment]
@ericnakagawa On it!

**Comment by lukemiles** [comment]
@ericnakagawa Test added.

File: Libraries/Components/WebView/WebView.android.js [hunk 1]
```diff
@@ -287,6 +287,22 @@ class WebView extends React.Component {
   };
 
   /**
+    * Injects a javascript string into the referenced
+    * WebView. Deliberately does not return a response
+    * because using eval() to return a response breaks
+    * this method on pages with a Content Security Policy
+    * that disallows eval(). If you need that functionality,
+    * look into postMessage/onMessage.
+    */
```
Comment (yungsters) [inline Libraries/Components/WebView/WebView.android.js]: Nit: Can you fix the indentation here? (One too many spaces.)

File: Libraries/Components/WebView/WebView.ios.js [hunk 2]
```diff
@@ -490,6 +490,22 @@ class WebView extends React.Component {
   };
 
   /**
+    * Injects a javascript string into the referenced
+    * WebView. Deliberately does not return a response
+    * because using eval() to return a response breaks
+    * this method on pages with a Content Security Policy
+    * that disallows eval(). If you need that functionality,
+    * look into postMessage/onMessage.
+    */
```
Comment (yungsters) [inline Libraries/Components/WebView/WebView.ios.js]: Ditto.

File: Examples/UIExplorer/js/WebViewExample.js [hunk 3]
```diff
@@ -442,5 +475,9 @@ exports.examples = [
   {
     title: 'Mesaging Test',
     render(): ReactElement<any> { return <MessagingTest />; }
-  }
+  },
+  {
+    title: 'Inject JavaScript',
+    render(): React.Element<any> { return <InjectJS/>; }
```
Comment (yungsters) [inline Examples/UIExplorer/js/WebViewExample.js]: Nit: Space before `/>`.

**Comment by mmmulani** [comment]
iOS side lgtm

**Comment by lukemiles** [comment]
@yungsters:

[screen shot 2016-12-12 at 5 19 05 pm](https://cloud.githubusercontent.com/assets/1691179/21123713/41ec3eb0-c08f-11e6-91a3-7a06adfb563a.png)

Managed to get it working on Android and fixed all the nits. Let me know what else this needs before it can be merged!

**Comment by lukemiles** [comment]
@ericnakagawa Anything else this needs before it can be merged?

**Comment by SRandazzo** [comment]
Just what I came here looking for! Thanks @lukemiles for this, looking forward to it landing in master

**Comment by lukemiles** [comment]
Thanks @SRandazzo!

@jacobp100 @yungsters @ericnakagawa What more has to be done here to get this merged into master?

**Comment by jacobp100** [comment]
Someone at Facebook has to merge it.

**Comment by lukemiles** [comment]
Made some stylistic changes thanks to suggestions - @brentvatne made. Would love to know if there's anything else blocking this PR!

**Comment by brentvatne** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**